File: Java/1020.java
__________________________________________________________________________________________________
sample 2 ms submission
class Solution {
    public int numEnclaves(int[][] A) {
        int result = 0;
        for(int i = 0; i < A.length; i++){
            dfs(A,i,0);
            dfs(A,i,A[0].length-1);
        } 
        for(int j = 0; j < A[0].length; j++) {
            dfs(A,0,j);
            dfs(A,A.length-1,j);
        }
        
        for(int i = 0; i < A.length; i++) 
            for(int j = 0; j < A[i].length; j++) 
                    result+=A[i][j];
        
        return result;
    }
    
    public void dfs(int a[][], int i, int j) {
        if(i >= 0 && i <= a.length - 1 && j >= 0 && j <= a[i].length - 1 && a[i][j] == 1) {
            a[i][j] = 0;
            dfs(a, i + 1, j);
            dfs(a, i - 1, j);
            dfs(a, i, j + 1);
            dfs(a, i, j - 1);
        }
    }
}
__________________________________________________________________________________________________
sample 46468 kb submission
class Solution {
    private int[][] dirs = new int[][]{{0,1},{0,-1},{1,0},{-1,0}};
    private void dfs(int[][] A,int i,int j){
        if(i < 0 || i >= A.length || j < 0 || j >=A[0].length) return;
        if(A[i][j] == 0) return;

        A[i][j] = 0;
        for(int[] d : dirs){
            dfs(A,i + d[0],j + d[1]);
        }
    }

    public int numEnclaves(int[][] A) {
        for (int i = 0; i < A.length; i++) {
            dfs(A,i,0);
            dfs(A,i,A[0].length - 1);
        }

        for (int i = 0; i < A[0].length; i++) {
            dfs(A,0,i);
            dfs(A,A.length - 1,i);
        }

        int ones = 0;
        for (int i = 0; i < A.length; i++) {
            for (int j = 0; j < A[0].length; j++) {
                if(A[i][j] == 1) ones++;
            }

        }
        return ones;
    }
}
__________________________________________________________________________________________________

File: Python3/1020.py
__________________________________________________________________________________________________
sample 536 ms submission
class Solution:
    def numEnclaves(self, A: List[List[int]]) -> int:
        
        def goDeep(row, col):
            if A[row][col]:
                A[row][col] = 0
                if row + 1 < len(A):
                    if A[row+1][col]:
                        goDeep(row+1, col)
                if row > 0:
                    if A[row-1][col]:
                        goDeep(row-1, col)
                if col + 1 < len(A[0]):
                    if A[row][col+1]:
                        goDeep(row, col+1)
                if col > 0:
                    if A[row][col-1]:
                        goDeep(row, col-1)
        
        for i in range(len(A)):
            goDeep(i, 0)
            goDeep(i, len(A[0])-1)
        for i in range(len(A[0])):
            goDeep(0, i)
            goDeep(len(A)-1, i)
            
        return sum([sum(i) for i in A])
__________________________________________________________________________________________________
sample 14112 kb submission
class Solution:
    def numEnclaves(self, A: List[List[int]]) -> int:
        if len(A) == 0 or len(A[0]) == 0:
            return 0
        
        m, n = len(A), len(A[0])
        
        directions = [(-1, 0), (0, 1), (1,0), (0,-1)]
        
        def dfs(i, j, A, m, n):
            if i < 0 or i >= m or j < 0 or j >= n or A[i][j] != 1:
                return
            A[i][j] = -1
            for dx, dy in directions:
                dfs(i + dx, j + dy, A, m, n)
                
        for i in range(m):
            dfs(i, 0, A, m, n)
            dfs(i, n-1, A, m, n)
            
        for j in range(1, n - 1):
            dfs(0, j, A, m, n)
            dfs(m-1, j, A, m, n)
            
        count = 0
        for i in range(m):
            for j in range(n):
                count += A[i][j] == 1
                    
        return count
__________________________________________________________________________________________________
